lib/logger.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var merge = require('lodash.merge');
+var util = require('util');
+var winston = require('winston');
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger.log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
+ *
+ * @param {Object} options - Configuration options.
+ * @return {Object} An instance of Winston.
+ */
+exports.logger = function (options) {
+  'use strict';
+
+  options = getOptions(options);
+
+  var exitOnError = options.logger.exitOnError;
+  var level = options.logger.level;
+  var logFile = options.logger.logFile;
+  var silent = options.logger.silent;
+  var transports = [
+    new winston.transports.Console({
```

Comment:
@scottnath
cc: @rebmullin @sergesemashko @conortm @gvorbeck @beynya @nikkiana
Ok. I appreciate the explanation, although my immediate reaction is that we want to create an APPLICATION, and not a bunch of plugins. You run into the problem of becoming Drupal or Gulp if you go too far into modularity.

I feel like this instance of extreme modularity might be trying to solve a problem that doesn't exist. Or doesn't concern us. Drupal and Gulp solve actual problems that require them being what they are. What is the problem we're trying to solve? Devs hate writing the same markup, css, and js over and over? Or maybe the expense of that time? That's what I heard is the problem.